core/snapshot/databases/metadata/database.go

```diff
@@ -0,0 +1,60 @@
+package metadata
+
+import (
+	"strconv"
+
+	"code.vegaprotocol.io/vega/libs/proto"
+
+	tmtypes "github.com/tendermint/tendermint/abci/types"
+)
+
+type Adapter interface {
+	Save(version []byte, state []byte) error
+	Load(version []byte) (state []byte, err error)
+	Close() error
+	Clear() error
+}
+
+type Database struct {
+	adapter Adapter
```

Comment:
nit-pick, but can't we just embed the adapter here? Saves us from having to add the `Close` and `Clear` functions

Reply:
Sure thing